src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1297,7 +1273,13 @@ void ImportStoreIndirect(TypeDesc type)
             var address = Pop();
 
             Check(!address.IsReadOnly, VerifierError.ReadOnlyIllegalWrite);
-            CheckIsAddressCompatibleWithType(address, type);
+
+            CheckIsByRef(address);
+            if (type != null)
+                CheckIsAssignablePointer(type, address.Type);
```

Comment:
The error messages from `CheckIsAssignablePointer` look pretty confusing. I do not remember why I have done them that way.

I am wondering whether we should delete `CheckIsAssignablePointer` and just call `CheckIsAssignable`. It seems to be doing the same thing.